### PR TITLE
Adding PollFile model

### LINKED ISSUE #51: Adding PollFile

### PR DESCRIPTION
**Pull request by OmerCS8**
- Adding PollFile model that stores files that belong to a Poll
- Adding matching tests

fix #51
Dependent on #49

### CODE REVIEW

File: posts/tests.py [hunk 1]
```diff
@@ -1,3 +1,88 @@
...
+# Check if the Poll can be referred as post.
+@pytest.mark.django_db()
+def test_poll_is_post(new_poll):
+    assert len(Post.objects.all()) == 0
+    new_poll.author.save()
+    new_poll.save()
+    assert len(Post.objects.all()) == 1
+    post = Post.objects.all().first()
+    assert hasattr(post, 'poll') is True
+    assert hasattr(post, 'resume') is False
```
Comment (abiton1) [inline posts/tests.py]:
```suggestion
    assert hasattr(post, 'poll')
    assert not hasattr(post, 'resume')
```
It is redundant to check if True or False

File: posts/tests.py [hunk 2]
```diff
@@ -19,70 +19,147 @@ def new_user():
...
+    def test_persist_poll(self, persist_Poll):
+        assert persist_Poll in Poll.objects.all()
+
+    # Check if the Poll can be referred as post and accessible from Post list .
+    def test_poll_is_post(self, persist_Poll):
+        assert isinstance(persist_Poll, Post)
+        assert len(Post.objects.all()) == 1
+        post = Post.objects.all().first()
+        assert hasattr(post, 'poll') is True
+        assert hasattr(post, 'resume') is False
```
Comment (abiton1) [inline posts/tests.py:74]:
```suggestion
       assert hasattr(post, 'poll')
       assert not hasattr(post, 'resume')
```

File: posts/tests.py [hunk 3]
```diff
@@ -19,70 +19,147 @@ def new_user():
...
+
+
 @pytest.fixture
 def new_poll(new_user):
     return Poll(author=new_user, description=DESCRIPTION)
 
 
-# -------------------------------------------- Poll tests --------------------------------------------
+@pytest.fixture
+def persist_Poll(new_poll):
```
Comment (abiton1) [inline posts/tests.py:34]: Use lowercase names `poll` instead of `Poll`

File: posts/tests.py [hunk 4]
```diff
@@ -19,70 +19,147 @@ def new_user():
...
+    new_poll.save()
+    return new_poll
 
 
-# Check that the Poll values are the same as the Poll inputs.
-def test_new_poll(new_poll):
-    assert new_poll.author.username == USERNAME
-    assert new_poll.description == DESCRIPTION
+@pytest.fixture
+def new_pollFile(new_poll):
```
Comment (abiton1) [inline posts/tests.py:41]: Use lowercase names `poll_file` instead of `pollFile`

File: posts/tests.py [hunk 5]
```diff
@@ -19,70 +19,147 @@ def new_user():
...
-    new_poll.author.save()
-    new_poll.save()
-    assert len(Post.objects.all()) == 1
-    post = Post.objects.all().first()
-    assert hasattr(post, 'poll') is True
-    assert hasattr(post, 'resume') is False
-    poll = post.poll
-    assert new_poll == poll
+@pytest.fixture
+def persist_pollFile(new_pollFile):
```
Comment (abiton1) [inline posts/tests.py:46]: `persist_poll_file` instead of `persist_pollFile`

File: posts/tests.py [hunk 6]
```diff
@@ -281,3 +295,66 @@ def test_delete_polls_author(self, persist_poll):
...
+@pytest.mark.django_db()
+class TestPollFile:
+    # Check that the PollFile values are the same as the PollFile inputs.
+    def test_new_poll_file_input_same_as_output(self, new_poll_file):
+        assert new_poll_file.poll.author.username == USERNAME
+        assert new_poll_file.poll.description == DESCRIPTION
+        assert new_poll_file.file == FILE1
+
+    # Check if the PollFile is saved in the database and accessible via its Poll.
+    @pytest.mark.django_db()
```
Comment (MatanP12) [inline posts/tests.py:311]: I don't need there is a need to add ` @pytest.mark.django_db()` annotation  on the methods, only on the class

File: posts/tests.py [hunk 7]
```diff
@@ -281,3 +295,66 @@ def test_delete_polls_author(self, persist_poll):
...
+    def test_persist_poll_file(self, persist_poll_file):
+        assert persist_poll_file in PollFile.objects.all()
+        assert persist_poll_file in persist_poll_file.poll.pollfile_set.all()
+
+    # Check if PollFile deletion delete only PollFile from database.
+    @pytest.mark.django_db()
```
Comment (MatanP12) [inline posts/tests.py:317]: same

File: posts/tests.py [hunk 8]
```diff
@@ -281,3 +295,66 @@ def test_delete_polls_author(self, persist_poll):
...
+        assert persist_poll_file.poll in Poll.objects.all()
+        assert persist_poll_file.poll.author in User.objects.all()
+        assert persist_poll_file in PollFile.objects.all()
+        persist_poll_file.delete()
+        assert persist_poll_file.poll in Poll.objects.all()
+        assert persist_poll_file.poll.author in User.objects.all()
+        assert persist_poll_file not in PollFile.objects.all()
+
+    # Check if PollFile's poll deletion delete both PollFile and poll from database - but not the poll's author.
+    @pytest.mark.django_db()
```
Comment (MatanP12) [inline posts/tests.py:328]: same(you got the idea)

File: posts/tests.py [hunk 9]
```diff
@@ -281,3 +295,66 @@ def test_delete_polls_author(self, persist_poll):
...
+    def test_several_pollfiles(self, persist_poll):
+        file1 = PollFile(poll=persist_poll, file=FILE1)
+        assert len(PollFile.objects.all()) == 0
+        file1.save()
+        assert len(PollFile.objects.all()) == 1
+        persist_poll.pollfile_set.create(file=FILE2)
+        assert len(persist_poll.pollfile_set.all()) == 2
+
+    # verify that it is impossible to create a PollFile with invalid params.
```
Comment (MatanP12) [inline posts/tests.py:353]: Can you specify what are the invalid params?

File: posts/tests.py [hunk 10]
```diff
@@ -281,3 +295,66 @@ def test_delete_polls_author(self, persist_poll):
...
+    # Check if PollFile's poll deletion delete both PollFile and poll from database - but not the poll's author.
+    @pytest.mark.django_db()
+    def test_delete_pollfile_poll(self, persist_poll_file):
+        persist_poll_file.poll.delete()
+        assert persist_poll_file.poll not in Poll.objects.all()
+        assert persist_poll_file.poll.author in User.objects.all()
+        assert persist_poll_file not in PollFile.objects.all()
+
+    # Check if PollFile's poll's author deletion delete also PollFile and database.
+    @pytest.mark.django_db()
```
Comment (MatanP12) [inline posts/tests.py:336]: .

File: posts/tests.py [hunk 11]
```diff
@@ -281,3 +295,66 @@ def test_delete_polls_author(self, persist_poll):
...
+    def test_delete_pollfile_user(self, persist_poll_file):
+        persist_poll_file.poll.author.delete()
+        assert persist_poll_file.poll not in Poll.objects.all()
+        assert persist_poll_file.poll.author not in User.objects.all()
+        assert persist_poll_file not in PollFile.objects.all()
+
+    # Check if Poll can have several PollFiles simultaneously.
+    @pytest.mark.django_db()
```
Comment (MatanP12) [inline posts/tests.py:344]: .

File: posts/tests.py (hunk 6 above)
Comment (yulisuliman) [reply posts/tests.py:311]: > I don't need there is a need to add ` @pytest.mark.django_db()` annotation on the methods, only on the class

I Agree

**Review by MatanP12 — APPROVED** [review]
LGTM Great work!

**Review by yulisuliman — DISMISSED** [review]

**Review by TomerNewmanPrograms — DISMISSED** [review]
Looks great!

**Review by abiton1 — APPROVED** [review]

**Comment by natifridman** [comment]
@OmerCS8 Please fix the conflict so we can merge this

**Review by MatanP12 — APPROVED** [review]
LGTM

**Comment by OmerCS8** [comment]
> @OmerCS8 Please fix the conflict so we can merge this

All conflicts are solved, and all commits are squashed. 👍

**Review by yulisuliman — APPROVED** [review]

**Review by MatanP12 — APPROVED** [review]

**Review by natifridman — APPROVED** [review]

**Pull request merged**